### ISSUE TITLE
TypeScript typings

### ISSUE THREAD

**Issue by AlexZeitler** [vweevers/node-docker-machine#27]
Thanks for this great package. 

Would you accept a PR providing TypeScript typings?

**Comment by vweevers** [comment]
Yes, but see the caveat I mentioned here: https://github.com/vweevers/node-docker-machine/issues/16#issuecomment-436181745

**Comment by AlexZeitler** [comment]
If it would be ok for you to ping me in case of upcoming API changes, I would prefer to add/maintain them with the package directly instead of DT.

**Comment by vweevers** [comment]
Alright!
